sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/Transforms.java

```diff
@@ -282,8 +281,21 @@ private static FormField setFormField(String name, FieldData valueData, FieldVal
             default:
                 throw LOGGER.logExceptionAsError(new RuntimeException("FieldValue Type not supported"));
         }
-        return new FormField(name, null, valueData, value,
-            setDefaultConfidenceValue(fieldValue.getConfidence()));
+
+        FieldData labelData = null;
+        // if 'name' is same as the raw field value text, it means this FormField is not a label.
+        // Otherwise, the FormField is for label, which might have label data, such as bounding box values.
+        if (name != null && name.equals(fieldValue.getText())) {
+            valueData = new FieldData(name, toBoundingBox(fieldValue.getBoundingBox()),
```

Comment:
from what I can see the method that calls `setFormField`  is always setting `valueData` but here you are assigning it again

Reply:
we should be consistent and only have a place where this logic happens or debbuging later will be a problem

Reply:
previous impl is confuse. Updated to only adding valueData for non-ARRAY type when valueData is null

Reply:
@mssfang Should this be enough to determine valueData=null for Array type field element https://github.com/Azure/azure-sdk-for-java/blob/62a9813e3865325e975551f1605a4a272abce7f9/sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/Transforms.java#L210

Reply:
the reason of having assign value of valueData is because valueData is always null. Let me change this section of code to   upper caller, 
```
    private static List<FormField> toFieldValueArray(List<FieldValue> valueArray, List<ReadResult> readResults) {
        return valueArray.stream()
            .map(fieldValue -> {
                FieldData valueData = null;
                // ARRAY has ho value data, such as bounding box.
                if (ARRAY != fieldValue.getType()) {
                    valueData = new FieldData(fieldValue.getText(), toBoundingBox(fieldValue.getBoundingBox()),
                        fieldValue.getPage() == null ? -1 : fieldValue.getPage(),
                        setReferenceElements(fieldValue.getElements(), readResults));
                }
                return setFormField(null, valueData, fieldValue, readResults);
            })
            .collect(Collectors.toList());
    }
```